### PR TITLE
Read from 'polymer.json' for project config

### PR DESCRIPTION
**Pull request by azakus**
Flags will override

### CODE REVIEW

**Comment by azakus** [comment]
@justinfagnani PTAL

File: src/project-config.ts [hunk 1]
```diff
@@ -24,8 +24,35 @@ export class ProjectConfig {
   shell: string;
   entrypoints: string[];
 
-  constructor(options: ProjectConfigOptions) {
+  static fromConfigFile(filepath: string): ProjectConfig {
+    try {
+      let configContent = fs.readFileSync(filepath, 'utf-8');
```
Comment (justinfagnani) [inline src/project-config.ts]: is there really no way to check a file besides catching an exception? `statSync`?

File: src/project-config.ts [hunk 2]
```diff
@@ -24,8 +24,35 @@ export class ProjectConfig {
...
+    this.init(configFile, options);
+  }
+
+  init(configFile?: string, options?: ProjectConfigOptions) {
     options = options || {};
+    if (configFile) {
+      // config file is default, options will override
+      let fromFile = ProjectConfig.fromConfigFile(configFile);
+      if (fromFile) {
+        for (let option in options) {
```
Comment (justinfagnani) [inline src/project-config.ts]: `Object.keys(options)` so you don't get things from the proto chain.

Comment (justinfagnani) [inline src/project-config.ts]: Actually, just use `Object.assign`

**Comment by justinfagnani** [comment]
LGTM, with nits.

File: src/project-config.ts (hunk 1 above)
Comment (azakus) [inline src/project-config.ts]: Nope. `fs.exists` is super deprecated.

File: src/project-config.ts (hunk 2 above)
Comment (azakus) [inline src/project-config.ts]: I was worried about Node v4 support, but let's let Travis tell us.

**Pull request merged**